GUIDE.md

```diff
@@ -0,0 +1,1081 @@
+ -V, --verbose            Enable verbose mode.
+```
+
+The `-b` option should be appended if you want stack traces that map to the correct lines of code of Roc `src/`. This is useful for debugging errors that originate from any Roc internals.
+
+The `-V` option should be appended for very verbose feedback of what is currently happening. This includes useful information like:
+
+- Extensions used
+  - The Roc extensions that are loaded in your current context
+- Extensions tree
+  - A visual presentation of the extension hierarchy loaded in your current context
+- Feedback on all hooks that are executed in Roc
+  - Gives a complete overview of what extensions that are executing an action on given hook
+
+If you are having a problem with a command and feel like you do not know where to begin, running it with `-b -V` will likely set you on the correct path.
+
+In addition, [debug](https://github.com/visionmedia/debug) is used internally. This means that you can use the `DEBUG` environment variable to access useful information. For example: `$ DEBUG="*" roc build`
+
+There are also these useful settings if you want to add this to your configuration:
+
+```
+--runtime-debug-client    Filter for debug messages that should be shown for the client, see https://npmjs.com/package/debug.
+ --runtime-debug-server   Filter for debug messages that should be shown for the server, see https://npmjs.com/package/debug.
+```
+
+For example: `$ roc start server --runtime-debug-server="koa:*"`
+
+If you believe it is not a configuration error on your side and you do not know how to fix the problem, the output here can be shared in an [Issue](https://github.com/rocjs/roc/issues). We are here to help.
+
+For a better understanding of the underlying architecture of Roc you can [read more about that in the docs](https://github.com/rocjs/roc/tree/master/docs#table-of-contents).
+
+### Start without CLI
+You can start your application [without having the CLI installed globally](https://github.com/rocjs/roc/blob/master/docs/Runtime.md#how-can-i-start-my-roc-application-without-using-the-roc-cli).
+
+### Dev vs build + start
+
+Please note that `roc dev` and `roc build && roc start` are two **very** different things.
+
+`dev` mode provides some extra useful utilities for discovering problems:
+- [YellowBox](https://github.com/iamdustan/yellowbox-react) is installed and active for notifying about potential issues
+- [Redux devtools](https://github.com/gaearon/redux-devtools) are available on `ctrl+h` by default
+- [Browsersync](https://www.browsersync.io/) admin tool is running on port 3031 by default
+- React is in development mode for better feedback
+- It enables hot-reloading of your code
+- It is unsuitable for production, in every way imaginable
+
+Never run `dev` in a production environment. It is tailored for a good development experience but is not suited for production traffic.
+
+Use `roc start` to serve your production-grade bundles.
```

Comment:
Give an example on how it's possible to create a custom entry point, for when using together with something like a logger on the server.